spec/Infrastructure/Serializer/ProductVariantNormalizerSpec.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\PriceHistoryPlugin\Infrastructure\Serializer;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\ApiBundle\SectionResolver\AdminApiSection;
+use Sylius\Bundle\ApiBundle\SectionResolver\ShopApiSection;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\PriceHistoryPlugin\Application\Provider\ProductVariantPriceProviderInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+final class ProductVariantNormalizerSpec extends ObjectBehavior
+{
+    function let(ProductVariantPriceProviderInterface $priceProvider, SectionProviderInterface $sectionProvider): void
+    {
+        $this->beConstructedWith($priceProvider, $sectionProvider);
+    }
+
+    function it_supports_only_product_variant_interface(ProductVariantInterface $variant, OrderInterface $order): void
+    {
+        $this->supportsNormalization($variant)->shouldReturn(true);
+        $this->supportsNormalization($order)->shouldReturn(false);
+    }
+
+    function it_supports_normalization_if_section_is_shop_get(
+        SectionProviderInterface $sectionProvider,
+        ProductVariantInterface $variant,
+        ShopApiSection $shopApiSection,
+    ): void {
+        $sectionProvider->getSection()->willReturn($shopApiSection);
+        $this->supportsNormalization($variant)->shouldReturn(true);
+    }
+
+    function it_does_not_support_if_section_is_admin_get(
+        SectionProviderInterface $sectionProvider,
+        AdminApiSection $adminApiSection,
+        ProductVariantInterface $variant,
+    ): void {
+        $sectionProvider->getSection()->willReturn($adminApiSection);
+        $this->supportsNormalization($variant)->shouldReturn(false);
+    }
+
+    function it_does_not_support_if_the_normalizer_has_been_already_called(ProductVariantInterface $variant): void
+    {
+        $this
+            ->supportsNormalization($variant, null, ['sylius_price_history_product_variant_normalizer_already_called' => true])
+            ->shouldReturn(false)
+        ;
+    }
+
+    function it_serializes_product_variant_if_item_operation_name_is_different_that_admin_get(
```

Comment:
I don't see any reference to an operation name inside the test
```suggestion
    function it_adds_lowest_price_before_discount_to_variant_data(
```
Also, the case where the value is null is missing